persist/file-adapter/adapter.go

```diff
@@ -85,18 +84,12 @@ func (a *Adapter) loadPolicyFile(model model.Model, handler func(string, model.M
 	}
 	defer f.Close()
 
-	buf := bufio.NewReader(f)
-	for {
-		line, err := buf.ReadString('\n')
-		line = strings.TrimSpace(line)
+	scanner := bufio.NewScanner(f)
```

Comment:
I took the opportunity to integrate my streamlined handler into the base file adapter, as it will be simpler for others to use as a reference IMO.